### ISSUE TITLE
[Bug]: Qwen3 模型在 Ascend 910B 上 temperature=0 时推理结果不确定

### ISSUE THREAD

**Issue by LORDVINCE** [vllm-project/vllm-ascend#9373]
### 环境信息

- vllm-ascend: v0.18.0
- 硬件: Ascend 910B (64GB HBM)
- CANN: 8.0.0
- 模型: Qwen3-4B (FP16)

### 

### 🐛 Describe the bug

**Labels:** bug

### 描述

在 Ascend 910B 上使用 vllm-ascend v0.18.0 部署 Qwen3-4B 模型，设置 `temperature=0`（贪心解码）时，对相同输入多次推理，输出结果不一致。

### 复现步骤

```bash
# 启动服务
python -m vllm.entrypoints.openai.api_server \
  --model Qwen/Qwen3-4B \
  --tensor-parallel-size 1 \
  --gpu-memory-utilization 0.9 \
  --dtype float16
```

```python
import openai
client = openai.Client(base_url="http://localhost:8000/v1", api_key="empty")

prompt = "请列举5个中国历史文化名城，并简要说明原因。"
for i in range(3):
    resp = client.chat.completions.create(
        model="Qwen/Qwen3-4B",
        messages=[{"role": "user", "content": prompt}],
        temperature=0.0,
        max_tokens=256,
    )
    print(f"--- Run {i+1} ---")
    print(resp.choices[0].message.content)
```

### 期望行为

`temperature=0` 时应为确定性输出，相同输入每次结果完全一致。

### 实际行为

多次运行产生不同的输出内容（非逐 token 完全相同），部分情况下最后一个城市名称或原因描述不同。

### 环境信息

- vllm-ascend: v0.18.0
- 硬件: Ascend 910B (64GB HBM)
- CANN: 8.0.0
- 模型: Qwen3-4B (FP16)

### 补充说明

FAQ 中提到需要设置多个环境变量才能保证确定性：

```bash
LCCL_DETERMINISTIC=1
HCCL_DETERMINISTIC=true
ATB_MATMUL_SHUFFLE_K_ENABLE=0
ATB_LLM_LCOC_ENABLE=0
```

建议：

1. 在文档中更醒目地标注 `temperature=0` 不等于确定性输出的限制
2. 或者在 `temperature=0` 时自动启用这些环境变量对应的内部开关
3. 提供一个统一的 `--deterministic` 启动参数

---

## Issue 2: [Bug] vllm-ascend 进程异常退出后 NPU 显存未释放

**Labels:** bug

### 描述

vllm-ascend 服务进程因 OOM 或其他异常崩溃退出后，NPU 显存未被正确释放。通过 `npu-smi info` 可观察到显存占用仍然很高，导致无法重新启动新的推理服务。

### 复现步骤

1. 启动一个占用较高显存的模型服务：

```bash
python -m vllm.entrypoints.openai.api_server \
  --model Qwen/Qwen3-4B \
  --gpu-memory-utilization 0.95 \
  --max-model-len 32768
```

2. 发送大量并发请求触发 OOM：

```bash
# 使用 benchmark 或压测工具
```

3. 进程崩溃后检查 NPU 状态：

```bash
npu-smi info
```

### 期望行为

进程退出后（无论是正常退出还是异常崩溃），NPU 显存应被完全释放。

### 实际行为

`npu-smi info` 显示 NPU 显存占用仍在 40GB+，需要手动执行以下操作才能恢复：

```bash
# 查找残留进程
ps aux | grep vllm
# 或强制重置设备
npu-smi set -t device-reset -i 0
```

### 环境信息

- vllm-ascend: v0.18.0
- 硬件: Ascend 910B
- CANN: 8.0.0

### 补充说明

对比 GPU 上的 vLLM，CUDA 的显存管理在进程退出时会自动回收。建议：

1. 在 vllm-ascend 中注册信号处理器（SIGTERM/SIGINT），确保异常退出时调用 ACL 资源释放接口
2. 在文档中增加 NPU 显存排障指南，说明 `npu-smi` 清理命令

**Comment by earthmanylf** [comment]
请问启用确定性计算的环境变量后，在temperature=0时输出是否确定？其他建议已收到，我会反馈给资料同事，审视相关资料

**Comment by veropeaky** [comment]
怎么可能是temperature=0就是确定性输出，这只是说每次输出采样的时候对logits启用贪婪算法，取得概率最高的类而非范围内随机采样。但是，本身vllm是动态batch推理的，而且没有固定seed、torch本身也不是固定的，无法保证每一次推理出来的logits是一样的

**Comment by sleepy-dev-bin** [comment]
> 请问启用确定性计算的环境变量后，在temperature=0时输出是否确定？其他建议已收到，我会反馈给资料同事，审视相关资料

即使开启确定性计算环境变量
LCCL_DETERMINISTIC=1
HCCL_DETERMINISTIC=true
ATB_MATMUL_SHUFFLE_K_ENABLE=0
ATB_LLM_LCOC_ENABLE=0

使用quay.io/ascend/vllm-ascend:nightly-releases-v0.22.1rc 镜像在A2机器启动dsv4-flash，max_numseqs=16

同样的请求(temperature=0),并发20请求时，偶现几次结果不一致。

**Comment by earthmanylf** [comment]
由于调度层面不可能做到batch的固定，所以结果出现不一致是非常有可能出现的情况。另外matmul相关的确定性计算环境变量应该是CLOSE_MATMUL_K_SHIFT=1
